### ISSUE TITLE
[UnnecessaryAbstractClass] False Negative reported for valid abstract class

### ISSUE THREAD

**Issue by keerthikorrapati** [detekt/detekt#5891]
Detekt falsely reports with the below error for a valid abstract class.

## Expected Behavior
If an abstract class with no body inherits from another abstract class, detekt should not report it as an UnnecessaryAbstractClass.
For example,

```kotlin
abstract class A :  B()
```

B must be an abstract/open class itself because it ends with () , which wouldn't be there for an interface

## Observed Behavior
For an empty abstract class A, detekt reports this as UnnecessaryAbstractClass, asking us to convert it into an interface instead.

```kotlin
abstract class A :  B()
```

## Your Environment
* Version of detekt used: 1.22.0

**Comment by cortinico** [comment]
We have a test for exactly your scenario:
https://github.com/detekt/detekt/blob/0123e2950137daf75840816435bed989820e5f72/detekt-rules-style/src/test/kotlin/io/gitlab/arturbosch/detekt/rules/style/UnnecessaryAbstractClassSpec.kt#L82-L86

> For an empty abstract class A, detekt reports this as UnnecessaryAbstractClass

Could you provide a bigger example?
I have the feeling you're getting a report for B instead.
Also are you using Type Resolution or not?

**Comment by BraisGabin** [comment]
How is `B` in your case? Couldn't `B` be converted to a interface and then implement A like this `interface A : B`?

This rule is opinated. It prefer the usage of `interface`s over `abstract class`es.

**Comment by keerthikorrapati** [comment]
We are talking about the example that I mentioned right? (Sorry the unit test and my example use A and B interchangeably)

For the use-case:
abstract class A :  B()
I cannot convert B into an interface as this is from an external library.

**Comment by schalkms** [comment]
Detekt does not and can not know about the external library you are using.
In this case, a suppress annotation is justified or path excludes could also work.
The rule works as expected, from my point of view.

**Comment by cckroets** [comment]
Can't we tell from the syntax: `B()` (notably the parenthesis), that `B` must be an `open class`, and therefore `A` cannot be an interface, even without type resolution?

**Comment by schalkms** [comment]
> Can't we tell from the syntax: `B()` (notably the parenthesis), that `B` must be an `open class`, and therefore `A` cannot be an interface, even without type resolution?

If it's a constructor call, then yes. In this case, the abstract class without abstract members shall be concrete (e.g. non-abstract). There's no need for type resolution. However, it's not possible to detect whether the called constructor comes from an external library. First, you'd have to define for a repo what an external library actually is.
